xdr/decorated_signature.go

```diff
@@ -0,0 +1,38 @@
+package xdr
+
+// NewDecoratedSignature constructs a decorated signature structure directly
+// from the given signature and key hint. Note that the key hint should
+// correspond to the key that created the signature, but this helper cannot
+// ensure that.
+func NewDecoratedSignature(sig []byte, keyHint [4]byte) DecoratedSignature {
+	return DecoratedSignature{
+		Hint:      SignatureHint(keyHint),
```

Comment:
❗  This is subtle, but the signer may not be a key.
```suggestion
// NewDecoratedSignature constructs a decorated signature structure directly
// from the given signature and hint. Note that the hint should
// correspond to the signer that created the signature, but this helper cannot
// ensure that.
func NewDecoratedSignature(sig []byte, hint [4]byte) DecoratedSignature {
	return DecoratedSignature{
		Hint:      SignatureHint(hint),
```

Reply:
Very subtle, thanks!